config/config.go

```diff
@@ -0,0 +1,141 @@
+}
+
+type AuthModule struct {
+	Type     string            `yaml:"type"`
+	UserPass *UserPassConfig   `yaml:"userpass,omitempty"`
+	APIKey   string            `yaml:"apikey,omitempty"`
+	AWS      *AWSConfig        `yaml:"aws,omitempty"`
+	TLS      *TLSConfig        `yaml:"tls,omitempty"`
+	Options  map[string]string `yaml:"options,omitempty"`
+}
+
+// AWSConfig contains settings for SigV4 authentication.
+type AWSConfig struct {
+	Region  string `yaml:"region"`
+	RoleARN string `yaml:"role_arn,omitempty"`
+}
+
+// TLSConfig allows per-target TLS options.
+type TLSConfig struct {
+	CAFile             string `yaml:"ca_file,omitempty"`
+	CertFile           string `yaml:"cert_file,omitempty"`
+	KeyFile            string `yaml:"key_file,omitempty"`
+	InsecureSkipVerify bool   `yaml:"insecure_skip_verify,omitempty"`
+}
+
+type UserPassConfig struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+// validate ensures every auth module has the required fields according to its type.
+func (c *Config) validate() error {
+	for name, am := range c.AuthModules {
+		// Validate fields based on auth type
+		switch strings.ToLower(am.Type) {
+		case "userpass":
+			if am.UserPass == nil || am.UserPass.Username == "" || am.UserPass.Password == "" {
+				return fmt.Errorf("auth_module %s type userpass requires username and password", name)
+			}
+		case "apikey":
+			if am.APIKey == "" {
+				return fmt.Errorf("auth_module %s type apikey requires apikey", name)
+			}
+		case "aws":
+			if am.AWS == nil || am.AWS.Region == "" {
+				return fmt.Errorf("auth_module %s type aws requires region", name)
+			}
+		case "tls":
+			// TLS auth type means client certificate authentication only (no other auth)
+			if am.TLS == nil {
+				return fmt.Errorf("auth_module %s type tls requires tls configuration section", name)
+			}
+			if am.TLS.CertFile == "" || am.TLS.KeyFile == "" {
+				return fmt.Errorf("auth_module %s type tls requires cert_file and key_file for client certificate authentication", name)
+			}
+			// Validate that other auth fields are not set when using TLS auth type
+			if am.UserPass != nil {
+				return fmt.Errorf("auth_module %s type tls cannot have userpass configuration", name)
+			}
+			if am.APIKey != "" {
+				return fmt.Errorf("auth_module %s type tls cannot have apikey", name)
+			}
+			if am.AWS != nil {
+				return fmt.Errorf("auth_module %s type tls cannot have aws configuration", name)
+			}
+		default:
+			return fmt.Errorf("auth_module %s has unsupported type %s", name, am.Type)
+		}
+
+		// Validate TLS configuration (optional for all auth types, provides transport security)
+		if am.TLS != nil {
+			// For cert-based auth (type: tls), cert and key are required
+			// For other auth types, TLS config is optional and used for transport security
+			if strings.ToLower(am.Type) == "tls" {
+				// Already validated above that cert and key are present
+			} else {
```

Comment:
This could be simplified
```suggestion
			if !strings.ToLower(am.Type) == "tls" {
```